src/platform/plugins/shared/discover/public/application/main/components/layout/cascaded_documents/cascaded_document_layout.tsx

```diff
@@ -0,0 +1,226 @@
+ */
+
+import React, { useMemo, useCallback, Fragment, useRef, useEffect } from 'react';
+import { useEuiTheme } from '@elastic/eui';
+import { type AggregateQuery } from '@kbn/es-query';
+import { type Filter } from '@kbn/es-query';
+import {
+  DataCascade,
+  DataCascadeRow,
+  DataCascadeRowCell,
+  type DataCascadeRowCellProps,
+} from '@kbn/shared-ux-document-data-cascade';
+import { RequestAdapter } from '@kbn/inspector-plugin/common';
+import type { UnifiedDataTableProps } from '@kbn/unified-data-table';
+import { getESQLStatsQueryMeta } from '@kbn/esql-utils';
+import { EsqlQuery } from '@kbn/esql-ast';
+import type { ESQLStatsQueryMeta } from '@kbn/esql-utils/src/utils/cascaded_documents_helpers';
+import { getStatsCommandToOperateOn } from '@kbn/esql-utils/src/utils/cascaded_documents_helpers';
+import type { DataTableRecord } from '@kbn/discover-utils';
+import { useDiscoverServices } from '../../../../../hooks/use_discover_services';
+import { useScopedServices } from '../../../../../components/scoped_services_provider/scoped_services_provider';
+import { useCurrentTabSelector, useAppStateSelector } from '../../../state_management/redux';
+import {
+  useEsqlDataCascadeRowHeaderComponents,
+  useEsqlDataCascadeHeaderComponent,
+  ESQLDataCascadeLeafCell,
+  type ESQLDataGroupNode,
+} from './blocks';
+import { cascadedDocumentsStyles } from './cascaded_documents.styles';
+import { type CascadedDocumentsRestorableState } from './cascaded_documents_restorable_state';
+import { useEsqlDataCascadeRowActionHelpers } from './blocks/use_row_header_components';
+import {
+  useDataCascadeRowExpansionHandlers,
+  useGroupedCascadeData,
+  useScopedESQLQueryFetchClient,
+} from './hooks';
+import type { DiscoverStateContainer } from '../../../state_management/discover_state';
+
+export { getESQLStatsQueryMeta };
+
+export interface ESQLDataCascadeProps extends Omit<UnifiedDataTableProps, 'ref'> {
+  defaultFilters?: Filter[];
+  viewModeToggle: React.ReactElement | undefined;
+  cascadeGroupingChangeHandler: (cascadeGrouping: string[]) => void;
+  cascadeConfig: CascadedDocumentsRestorableState;
+  togglePopover: ReturnType<typeof useEsqlDataCascadeRowActionHelpers>['togglePopover'];
+  queryMeta: ESQLStatsQueryMeta;
+  registerCascadeRequestsInspectorAdapter: (requestAdapter: RequestAdapter) => void;
+}
+
+const ESQLDataCascade = React.memo(
+  ({
+    rows,
+    dataView,
+    viewModeToggle,
+    cascadeConfig,
+    cascadeGroupingChangeHandler,
+    togglePopover,
+    queryMeta,
+    registerCascadeRequestsInspectorAdapter,
+    ...props
+  }: ESQLDataCascadeProps) => {
+    const [query, defaultFilters] = useAppStateSelector((state) => [state.query, state.filters]);
+    const [globalState, esqlVariables] = useCurrentTabSelector((state) => [
+      state.globalState,
+      state.esqlVariables,
+    ]);
+    const globalFilters = globalState?.filters;
+    const globalTimeRange = globalState?.timeRange;
```

Comment:
As mentioned in the previous comment https://github.com/elastic/kibana/pull/220119#discussion_r2509781341 using `dataRequestParams.timeRangeAbsolute` instead would help to eliminate differences between data in the group header and inside the group. Related comment: https://github.com/elastic/kibana/pull/220119#discussion_r2609115839